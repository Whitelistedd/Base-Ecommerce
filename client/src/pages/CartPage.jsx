import React from "react";
import { Cart } from "../components/Cart/Cart";
import { Layout } from "../components/Layout/Layout";

export const CartPage = () => {
  return (
    <Layout>
      <Cart />
    </Layout>
  );
};
