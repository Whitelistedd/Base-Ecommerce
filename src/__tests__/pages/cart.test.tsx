import { AnyAction, configureStore } from '@reduxjs/toolkit'

import CartPage from '@/pages/cart'
import { Provider } from 'react-redux'
import cartSlice from '@/redux/slices/cart'
import renderer from 'react-test-renderer'
import userSlice from '@/redux/slices/user'

jest.mock('next/router', () => require('next-router-mock'))

describe('Cart page', () => {
  const mockStore = configureStore({
    reducer: {
      user: userSlice,
      cart: cartSlice,
    },
  })
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore}>
          <CartPage />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
