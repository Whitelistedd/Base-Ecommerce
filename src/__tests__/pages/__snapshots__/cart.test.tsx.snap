// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart page should match snapshot 1`] = `
<div
  className="sc-f2b7308a-10 eHvoGd"
>
  <div
    className="sc-1a8f94cc-1 bgHcty"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-1ydvvti-MuiTypography-root"
    >
      YOUR CART IS EMPTY
    </p>
    <a
      href="/products"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium sc-d4e95e10-0 cscvRh sc-1a8f94cc-0 mnunh css-1vqlznj-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        data-testid="Button"
        disabled={false}
        id=":r0:"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        SHOP OUR PRODUCTS
      </button>
    </a>
  </div>
</div>
`;
